mmedit/models/backbones/sr_backbones/ttsr_net.py

```diff
@@ -210,3 +217,211 @@ def forward(self, x1, x2, x4):
         x = torch.clamp(x, -1, 1)
 
         return x
+
+
+@BACKBONES.register_module()
+class TTSRNet(nn.Module):
+    """TTSR network structure (main-net) for reference-based super-resolution.
+
+    Args:
+        in_channels (int): Number of channels in the input image
+        out_channels (int): Number of channels in the output image
+        mid_channels (int): Channel number of intermediate features.
+            Default: 64
+        num_blocks (tuple[int]): Block numbers in the trunk network.
+            Default: (16, 16, 8, 4)
+        res_scale (float): Used to scale the residual in residual block.
+            Default: 1.
+
+    """
+
+    def __init__(self,
+                 in_channels,
+                 out_channels,
+                 mid_channels=64,
+                 texture_channels=64,
+                 num_blocks=(16, 16, 8, 4),
+                 res_scale=1.0):
+        super().__init__()
+
+        self.texture_channels = texture_channels
+
+        self.sfe = SFE(in_channels, mid_channels, num_blocks[0], res_scale)
+
+        # stage 1
+        self.first1 = _conv3x3_layer(4 * texture_channels + mid_channels,
+                                     mid_channels)
+
+        self.rb1 = make_layer(
+            ResidualBlockNoBN,
+            num_blocks[1],
+            mid_channels=mid_channels,
+            res_scale=res_scale)
+
+        self.last1 = _conv3x3_layer(mid_channels, mid_channels)
+
+        # up-sampling 1 -> 2
+        self.up1 = PixelShufflePack(
+            in_channels=mid_channels,
+            out_channels=mid_channels,
+            scale_factor=2,
+            upsample_kernel=3)
+
+        # stage 2
+        self.first2 = _conv3x3_layer(2 * texture_channels + mid_channels,
```

Comment:
Always start with "conv" if it is conv. In this case, 

* first1 -> conv1
* first2 -> conv2

Reply:
Start with `conv` and end with `conv`, so named as `first` and `last`